shelley/chain-and-ledger/dependencies/non-integer/doc/non-integer-calculations.tex

```diff
@@ -75,124 +122,174 @@
 \section{Introduction}
 \label{sec:introduction}
 
-The Shelley specification~\cite{shelley_spec} and the Ouroboros
-protocol~\cite{ouroboros} use non-integral calculations. Most can be done in an
-exact way using rational numbers and arbitrary precision integers which are
-available in most programming languages or via external libraries.~\footnote{The
-  most widely used library is the GNU multi-precision library (GMP)}
+%% Actioned this. KH
+%\khcomment{Once this is distributed to a broader group, there should be some versioning and changelog mechanism (the PDF creation
+%  date isn't enough, since it doesn't have the source version).  This will help ensure that we're referring to the correct document :)
+%  I have some macros that could be adapted.}
 
-There are also instances where elementary functions are used which cannot be
-represented as rational numbers and therefore require a different
-treatment. This includes using the exponential function
-$e^{x}, x \in \mathbb{R}$ to calculate a decay and non-integral exponentiation
-$x^{y}, x, y \in \mathbb{R}$.
+The Shelley ledger specification~\cite{shelley_spec}\khcomment{I'd prefer a reference format that was shorter e.g. [G\"ud19], and made it clearer what were internal, public, published etc. documents - I've applied a possible format here.  We should also ideally use a single references.bib...} and the Ouroboros
+protocol~\cite{ouroboros} use non-integral calculations in a few places. Most of these can be
+implemented in an exact way using rational numbers and arbitrary precision integers, which are
+available either directly as programming language constructs (e.g. in Haskell) or via external libraries.~ \footnote{The
+  most widely used library is the GNU multi-precision library (GMP)}
+%
+In addition, there are some instances where the results of elementary functions cannot be
+precisely represented as rational numbers, and therefore require a representation that covers a wider
+range of \emph{real numbers}.
+These include the exponential function
+$e^{x}, x \in \mathbb{R}$, which is used to calculate how refunds decay over time, and general non-integral exponentiation
+$x^{y}, x, y \in \mathbb{R}$, which is used to \khcomment{MG, please fix!}.
```

Comment:
@kevinhammond  what is here to fix ? In a commit, you removed " ... calculate how refunds decay over time, and general non-integral exponentiation" why ?

Reply:
To make it clear to the reader why these are needed.

Reply:
You can expand if you think this is needed.